docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+        }
+    }
+}
+```
+
+Before pulling the container image layers, match the policy in the following order according to the location/registry of the container image and execute the corresponding policies:
+
+- Match single specific scopes under each member in transport, for example:
+  - For the image from the container registry that implements the Docker Registry HTTP API V2 protocol, match each scope under the "docker" member;
+  - For the image from local file system, match each scope under the "dir" member;
+- Match the default scope under the member in the transport. for example, the "" scope of the transport "dir".
+- Match the global default scope, such as the "default" field in the first line of the above example. We suggest that this policy must be configured to reject for security.
+
+After the matching is successful, the policy set under the corresponding scope is executed. In the design of confidential containers V1, only three policy types are currently supported now: "reject", "insecureAcceptAnything" and "signedBy". The former two is as the name implies, and the "signedBy" policy requires that there must exist at least one signature of this image can be verified by the public key ring provided by `keyPath` or `keyData` filed. In CCv1, he format of "signedBy" policy is defined as follows:
+
+```json
+{
+    "type":    "signedBy",
+    "keyType": "GPGKeys", /* The only currently supported value */
+    "keyPath": "/path/to/local/keyring/file", /* specifies the GPG Keyring that contains one or more public keys. */
+    /* Use one of the keyPath and keyData fields is enough */
+    "keyData": "base64_encoded_keyring_data",
+}
+```
+
+For the example of the policy file given above, the meaning of its security policy is as follows:
+
+1. For the image from the local file system, allowed to pull and run no matter there is a valid signature or not.
+
+2. For the image from the "docker.io/my_private_registry" registry, there must exist at least one signature that can be verified by the keys in keyData.
+
+3. For the image from the "registry.access.redhat.com", there must exist at least one signature that can be verified by the keys in `/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release`.
+
+4. Reject all other images.
+
+The above example provides a secure and robust policy configuration, but the policy file still provides sufficient configuration flexibility for weak security requirements. For example, if you need to completely disable the container image signature verification function, policy file can be:
+
+```json
+{
+  "default": [
+    {
+      "type": "insecureAcceptAnything"
+    }
+  ],
+}
+```
+
+We must remind that this configuration is very insecure. It will pull all unsigned container images. Or if you want to unconditionally accept an image from a specific registry, you just need to change the policy of the member matching the registry under the corresponding scope to `insecureAcceptAnything`.
```

Comment:
"remind" -> "stress" or "point out" or "highlight" or "warn"

Reply:
Note that the insecurity is actually somewhat limited for any image with at least one encrypted layer, even for a weak `policy.json`, since the decryption would fail unless the KBS handed over the keys, no?

Reply:
> Note that the insecurity is actually somewhat limited for any image with at least one encrypted layer, even for a weak `policy.json`, since the decryption would fail unless the KBS handed over the keys, no?

Exactly :)